lib/openssl/ssl.rb

```diff
@@ -276,7 +277,10 @@ def verify_certificate_identity(cert, hostname)
             if san.value.size == 4
               return true if san.value.unpack('C*').join('.') == hostname
             elsif san.value.size == 16
-              return true if san.value.unpack('n*').map { |e| sprintf("%X", e) }.join(':') == hostname
+              begin
+                return true if san.value == IPAddr.new(hostname).hton
+              rescue IPAddr::InvalidAddressError
+              end
```

Comment:
This works for both IPv4 and v6 addresses. And can you remove the misleading comment "follows GENERAL_NAME_print() [...]" while you are at it?

Reply:
Done.